samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -76,44 +75,33 @@
   private JobModel newJobModel;
   private int debounceTimeMs;
 
-  public ZkJobCoordinator(Config config, MetricsRegistry metricsRegistry) {
+  ZkJobCoordinator(Config config, MetricsRegistry metricsRegistry, ZkUtils zkUtils) {
     this.config = config;
-    ZkConfig zkConfig = new ZkConfig(config);
-    ZkKeyBuilder keyBuilder = new ZkKeyBuilder(new ApplicationConfig(config).getGlobalAppId());
-    ZkClient zkClient = ZkCoordinationServiceFactory
-        .createZkClient(zkConfig.getZkConnect(), zkConfig.getZkSessionTimeoutMs(), zkConfig.getZkConnectionTimeoutMs());
-
-    // setup a listener for a session state change
-    // we are mostly interested in "session closed" and "new session created" events
-    zkClient.subscribeStateChanges(new ZkSessionStateChangedListener());
```

Comment:
Looks like the session state change listener is no longer created in this patch. Was this accidentally removed ?